pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/PulsarAdminTool.java

```diff
@@ -157,6 +164,28 @@ public void setupCommands(Function<PulsarAdminBuilder, ? extends PulsarAdmin> ad
             for (Map.Entry<String, Class<?>> c : commandMap.entrySet()) {
                 addCommand(c, admin);
             }
+
+            CommandExecutionContext context = new CommandExecutionContext() {
+                @Override
+                public PulsarAdmin getPulsarAdmin() {
+                    return admin.get();
+                }
+
+                @Override
+                public Properties getConfiguration() {
+                    return properties;
+                }
+            };
+            loadCustomCommandFactories();
+
+            for (CustomCommandFactory factory : customCommandFactories) {
+                List<CustomCommandGroup> customCommandGroups = factory.commandGroups(context);
+                for (CustomCommandGroup group : customCommandGroups) {
+                    Object generated = CustomCommandsUtils.generateCliCommand(group, context, admin);
+                    jcommander.addCommand(group.name(), generated);
```

Comment:
what happen if you put two extensions that declares the same group ?

Reply:
it will override the other one.
not a big deal.
I don't think this will happen in the real life